lib/block_view_dao_coin_limit_order.go

```diff
@@ -620,18 +620,14 @@ func (bav *UtxoView) _connectDAOCoinLimitOrder(
 	} else {
 		desoAllowedToSpendByPublicKey[*NewPublicKey(txn.PublicKey)] = 0
 	}
+
 	// Iterate over all the UTXOs in the txn and spend them. As we spend each one,
 	// add the amount each account is allowed to spend to our map.
 	spentUtxoEntries := []*UtxoEntry{}
-	publicKeys := []PublicKey{}
-	for publicKey := range txMeta.MatchingBidsInputsMap {
-		publicKeys = append(publicKeys, publicKey)
-	}
-	sort.Slice(publicKeys, func(ii, jj int) bool {
-		return bytes.Compare(publicKeys[ii].ToBytes(), publicKeys[jj].ToBytes()) > 0
-	})
-	for _, publicKey := range publicKeys {
-		matchingBidsInputs := txMeta.MatchingBidsInputsMap[publicKey]
+	for _, transactor := range txMeta.MatchedBidsTransactors {
```

Comment:
I think we do still need to sort this slice somehow to make it deterministic (hypersync requirement). Let's sort by public key